pkg/streamingpromql/types/ring_buffer_test.go

```diff
@@ -563,3 +563,133 @@ func setupRingBufferTestingPools(t *testing.T) {
 		putHPointSliceForRingBuffer = originalPutHPointSlice
 	})
 }
+
+func TestFPointRingBufferEmptySet(t *testing.T) {
+	floats := NewFPointRingBuffer(&limiter.MemoryConsumptionTracker{})
+	require.NoError(t, floats.Use([]promql.FPoint{}))
+	it := NewFPointRingBufferViewIterator(floats.ViewAll(nil))
+	require.False(t, it.HasNext())
+	it.advance()
+	it.advance()
```

Comment:
It seems weird to be testing an internal method here - is there an exported function that could be used here?